GoogleInstagram/app/src/main/java/com/example/cynchen/googleinstagram/AllImagesFragment.java

```diff
@@ -0,0 +1,181 @@
+        //Three: k is at the first image of the list -> go to the last image
+        //Four: All other k index values -> load the (k-1)th image
+        previousbutton_all = (Button) rootView.findViewById(R.id.previous_image_all);
+        previousbutton_all.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (saved_images_all.size() == 0){
+                    create_alert_all_images(); //alert dialog creation function
+                }
+                else if (saved_images_all.size() == 1){
+                    webView_all.loadUrl(saved_images_all.get(k));
+                }
+                else if (k == 0){
+                    k = saved_images_all.size()-1;
+                    webView_all.loadUrl(saved_images_all.get(k));
+                }
+                else {
+                    k -= 1;
+                    webView_all.loadUrl(saved_images_all.get(k));
+                }
+            }
+        });
+
+        //Removing images: 1. remove from database, 2. remove from list of saved images
+        //One: 0 images saved -> popup alert dialog again
+        //Two: Removed image is the last image of the list -> remove that image from db and arraylist, then check if the
+        //new arraylist size is zero (in this case, load blank webview) or not (in this case, load the first image of the new list)
+        //Three: For all other k values -> delete from db and arraylist, then check if the new arraylist is zero (in this case, load blank
+        //webview) or not (in this case, load the new kth value of the shifted list)
+        remove_image = (Button) rootView.findViewById(R.id.remove_image_button);
+        remove_image.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (saved_images_all.size() == 0){
+                    create_alert_all_images();
+                }
+                else if (k == saved_images_all.size()-1){
+                    dbhelper.deleteData(saved_images_all.get(k));
+                    saved_images_all.remove(saved_images_all.get(k));
+                    if (saved_images_all.size() == 0) {
+                        webView_all.loadUrl("about:blank");
+                    }
+                    else {
+                        webView_all.loadUrl(saved_images_all.get(0));
+                    }
+                }
+                else {
+                    dbhelper.deleteData(saved_images_all.get(k));
+                    saved_images_all.remove(saved_images_all.get(k));
+                    if (saved_images_all.size() == 0) {
+                        webView_all.loadUrl("about:blank");
+                    }
+                    else {
+                        webView_all.loadUrl(saved_images_all.get(k));
+                    }
+                }
+            }
+        });
```

Comment:
move all of this button logic for all buttons out of onCreateView. This function should not be longer than your screen (you should be able to see the opening and closing {}s of the function on your screen.